app/api/chat/route.ts

```diff
@@ -22,13 +22,43 @@ const buildGoogleGenAIPrompt = (messages: Message[]) => ({
 export async function POST(req: Request) {
   const { messages } = await req.json();
 
+  return generateTestFile(messages);
+}
+
+export async function generateTestFile(messages: Message[]) {
+  const systemPrompt =
+    'Create 5 test cases in TypeScript using vitest based on the following requirements. Provide the full file.';
+
+  // Map messages to concatenate each message's content with the system prompt
+  const messagesWithPrompt = messages.map((message: any) => ({
+    role: message.role,
+    content: `${systemPrompt}\n${message.content}`,
+  }));
+
+  const geminiStream = await genAI
+    .getGenerativeModel({ model: 'gemini-pro' })
+    .generateContentStream(buildGoogleGenAIPrompt(messagesWithPrompt));
+
+  const stream = GoogleGenerativeAIStream(geminiStream);
+
+  return new StreamingTextResponse(stream);
+}
+
+export async function regenerateTestFile(
+  prevUserPrompt: string,
+  previousTestFile: string,
+  failureErrors: string
+) {
+  const systemPrompt =
+    'Create 5 different test cases in TypeScript using vitest based on the requirements for test cases, the previous test file, and the errors we encountered in the previous version. Provide the full file.';
+  const finalMsg = `${systemPrompt}\n//These were the requirements for the test cases.\n${prevUserPrompt}\n//This was the previous test file.\n${previousTestFile}\nThese were the errors we encountered in the previous test file:${failureErrors}`;
   const geminiStream = await genAI
     .getGenerativeModel({ model: 'gemini-pro' })
-    .generateContentStream(buildGoogleGenAIPrompt(messages));
+    .generateContentStream(
+      buildGoogleGenAIPrompt([{ role: 'assistant', content: finalMsg }])
+    );
```

Comment:
Looks good. I haven't test this one yet so we'll probably need some testing before we can merge this PR.